src/django/api/models.py

```diff
@@ -2959,6 +2959,16 @@ class Meta():
     )
     objects = TrainedModelManager()
 
+class dedupe_indexed_records(models.Model):
+    class Meta:
+        indexes = [
+            models.Index(fields=['block_key', 'record_id'],
+                         name='dedupe_indexed_records_idx')
+        ]
+    block_key = models.TextField(unique=True)
+    record_id = models.CharField(unique=True,
+                                 max_length=32)
```

Comment:
`block_key` and `record_id` are not unique. The `indexes` option Meta option will handle the unique pair.

```suggestion
    block_key = models.TextField()
    record_id = models.CharField(max_length=32)
```

This change will require deleting and recreating the migration with makemigrations.

Reply:
I had added a UniqueConstraint meta option before I saw your comment. Is there any reason to leave this along with the index, or is it not necessary to have both?
```
constraints = [UniqueConstraint(fields=['block_key',
                                                'record_id'],
                                        name='unique_blockkey_recordid')]
```

Reply:
Good catch. We do need to have this unique constraint.